### PR TITLE
feat(Pinia): migrate quoteReplyStore (Vuex) to chatExtrasStore (Pinia)

### LINKED ISSUE #11053: follow-up(quoteReplyStore) - migrate to Pinia
Tasks:
- [ ] migrate store to Pinia
  - [ ] add test coverage
- [ ] Issue with NcRichContenteditable parsing
  - [ ] move to utils
  - [ ] add test coverage

_Originally posted by @Antreesy in https://github.com/nextcloud/spreed/issues/11051#issuecomment-1833862257_

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #11053 

### 🖌️ UI Checklist

* No visual changes

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/stores/chatExtras.js [hunk 1]
```diff
@@ -56,14 +88,67 @@ export const useChatExtrasStore = defineStore('chatExtras', {
...
+		setChatInput({ token, text }) {
+			if (text !== null) {
+				// FIXME upstream: https://github.com/nextcloud-libraries/nextcloud-vue/issues/4492
+				const temp = document.createElement('textarea')
+				temp.innerHTML = text?.replace(/&/gmi, '&amp;') || ''
+				const parsedText = temp.value.replace(/&amp;/gmi, '&').replace(/&lt;/gmi, '<')
+					.replace(/&gt;/gmi, '>').replace(/&sect;/gmi, '§')
+
+				Vue.set(this.chatInput, token, parsedText)
+			} else {
```
Comment (Antreesy) [inline src/stores/chatExtras.js:138]: Maybe add `removeChatInput` action, as we touching it?

File: src/components/NewMessage/NewMessage.vue [hunk 2]
```diff
@@ -82,8 +82,8 @@
 						</template>
 					</NcButton>
 				</div>
-				<div v-if="messageToBeReplied" class="new-message-form__quote">
-					<Quote is-new-message-quote v-bind="messageToBeReplied" />
+				<div v-if="replyMessage" class="new-message-form__quote">
```
Comment (DorraJaouad) [inline src/components/NewMessage/NewMessage.vue:85]: eeh `parentMessage` maybe ?

File: src/store/conversationsStore.js [hunk 3]
```diff
@@ -317,10 +318,12 @@ const actions = {
 	 * Delete a conversation from the store.
 	 *
 	 * @param {object} context default store context;
-	 * @param {object} token the token of the conversation to be deleted;
+	 * @param {string} token the token of the conversation to be deleted;
 	 */
 	deleteConversation(context, token) {
 		// FIXME: rename to deleteConversationsFromStore or a better name
+		const chatExtrasStore = useChatExtrasStore()
+		chatExtrasStore.purgeChatExtras(token)
```
Comment (DorraJaouad) [inline src/store/conversationsStore.js:326]: Same for `clearMessagesHistory()` ?

Comment (Antreesy) [reply src/store/conversationsStore.js:326]: That would be an unexpected behaviour, if you click on 'Delete chat messages' and:
- Clear your input
- Remove absence status

File: src/stores/chatExtras.js [hunk 4]
```diff
@@ -25,11 +26,42 @@ import Vue from 'vue'
...
 
+/**
+ * @typedef {string} Token
+ */
+
+/**
+ * @typedef {object} State
+ * @property {{[key: Token]: object}} absence - The absence status per conversation.
+ * @property {{[key: Token]: number}} parentToReply - The parent message id to reply per conversation.
+ * @property {{[key: Token]: string}} chatInput -The input value per conversation.
```
Comment (DorraJaouad) [inline src/stores/chatExtras.js:37]:
```suggestion
 * @property {Object.<Token, Object>} absence - The absence status per conversation.
 * @property {Object.<Token, number>} parentToReply - The parent message id to reply per conversation.
 * @property {Object.<Token, string>} chatInput -The input value per conversation.
```

There is no syntax in JSDoc but the documentation state to define it like that 
https://jsdoc.app/tags-type

File: src/components/NewMessage/NewMessage.vue [hunk 5]
```diff
@@ -516,7 +516,7 @@ export default {
 
 			if (this.upload) {
 				// Clear input content from store
-				this.$store.dispatch('setCurrentMessageInput', { token: this.token, text: '' })
+				this.chatExtrasStore.setChatInput({ token: this.token, text: '' })
```
Comment (DorraJaouad) [inline src/components/NewMessage/NewMessage.vue:519]: Maybe it's better to remove it from the store completely and evaluate against its presence ? 
image

**Review by DorraJaouad — APPROVED** [review]
Nothing is blocking, all good 🦅.
But some tiny remarks

File: src/components/NewMessage/NewMessage.vue (hunk 5 above)
Comment (Antreesy) [reply src/components/NewMessage/NewMessage.vue:519]: Empty value is the default behaviour for every chat opening. So it doesn't make sense to delete reactive variable from the store and put it back again. In addition, we:ll never get a null as a value for NcRichContenteditable

Comment (DorraJaouad) [reply src/components/NewMessage/NewMessage.vue:519]: I checked, it's okay.

File: src/stores/chatExtras.js (hunk 4 above)
Comment (Antreesy) [reply src/stores/chatExtras.js:37]: image

Anyway, it is here just for reference, until we get Typescript (one day)

**Comment by Antreesy** [comment]
/backport to stable28

As it's tested fine on master, and may cause conflicts for further backports later on

**Pull request merged**